arch/arm/src/stm32h7/stm32_fdcan_sock.c

```diff
@@ -0,0 +1,2522 @@
+/****************************************************************************
+ * arch/arm/src/stm32h7/stm32_fdcan_sock.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+#include <unistd.h>
+#include <debug.h>
+#include <errno.h>
+
+#include <nuttx/can.h>
+#include <nuttx/wdog.h>
+#include <nuttx/irq.h>
+#include <nuttx/arch.h>
+#include <nuttx/wqueue.h>
+#include <nuttx/signal.h>
+#include <nuttx/net/netdev.h>
+#include <nuttx/net/can.h>
+#include <netpacket/can.h>
+
+#ifdef CONFIG_NET_CAN_RAW_TX_DEADLINE
+#include <sys/time.h>
+#endif
+
+#include <arch/board/board.h>
+
+#include "arm_internal.h"
+#include "chip.h"
+#include "stm32.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* General Configuration ****************************************************/
+
+#if !defined(CONFIG_SCHED_WORKQUEUE)
+#  error Work queue support is required; HPWORK is recommended
+#endif
+
+#ifdef CONFIG_NET_CAN_RAW_TX_DEADLINE
+#  define TX_TIMEOUT_WQ
+#endif
+
+/* If processing is not done at the interrupt level, then work queue support
+ * is required.
+ *
+ * The high-priority work queue is suggested in order to minimize latency of
+ * critical Rx/Tx transactions on the CAN bus.
+ */
+
+#define CANWORK HPWORK
```

Comment:
The network drivers should not run on the high priority work queue - https://cwiki.apache.org/confluence/display/NUTTX/Work+Queue+Deadlocks
Can we make this configurable with default LPWORK enabled ? Like in https://github.com/apache/incubator-nuttx/blob/fdef3a7b9220e53d898742251a6f5a721beb557b/arch/arm/src/stm32h7/stm32_ethernet.c#L101-L108

Reply:
Bu, the lp work will hunt performance in our test since it's priority is same as normal thread by default:
```
config SCHED_LPWORKPRIORITY
	int "Low priority worker thread priority"
	default 100
	---help---
		The minimum execution priority of the lower priority worker thread.
```

Reply:
Good point.  Implemented.

Reply:
There are many places in the code base, where it is clearly stated that "the network should NEVER run on the high priority work queue!". So this is more a change for consistency with other drivers. 

If the user wants to use HPWORK for the highest performance, then it should be clearly stated in the configuration. Then we can assume that the user knows what he or she is doing and is aware of potential drawbacks.

Reply:
Sure, just give some measurement on real device.